sqlmesh/core/snapshot/evaluator.py

```diff
@@ -898,6 +891,23 @@ def _create_schemas(self, tables: t.Iterable[t.Union[exp.Table, str]]) -> None:
             logger.info("Creating schema '%s'", schema)
             self.adapter.create_schema(schema)
 
+    def _table_deployability_flags(
+        self, snapshot: Snapshot, is_snapshot_deployable: bool
+    ) -> t.List[bool]:
+        flags = (
+            []
+            if (
+                is_snapshot_deployable
+                and not snapshot.reuses_previous_version
+                and not snapshot.is_managed
+            )
+            else [False]
+        )
+        if not snapshot.reuses_previous_version:
+            flags.append(True)
```

Comment:
To reiterate: we need to make sure we recreate the deployable table if it's actually missing.

Reply:
Refactored this to always check if the deployable table is missing and moved logic of identifying tables to be created to the `create` function. Then after getting the missing tables that need to be built, passing these table `versions_to_create` (based on deployability) to the `_create_snapshot` function to avoid duplicating logic